### PR TITLE
Finished docs for new Sage One element

### LINKED ISSUE #684: New Element: Sage One
## Description
New Element: Sage One V3 

## Notes
Reference the development stories. 
- https://github.com/cloud-elements/soba/issues/5944
- https://github.com/cloud-elements/soba/issues/5309

### PR DESCRIPTION
**Pull request by davewilks**
## Highlights
* New Sage One docs
*

## Closes
* Closes #684

### CODE REVIEW

**Comment by rsbarbo** [comment]
@davewilks @claytonshaver 

If I may jump in and provide some feedback in this PR, I think there are some additional details about Sage One that needs to be added to this documentation that has not yet been done. An example that I would like to highlight is the fact that customer needs to visit a few different websites to different pieces of information (i.e. https://developer.columbus.sage.com/ **vs.** https://developer.sageone.com/). We have a lot of great information, but Sage One latest release does not have the most intuitive flow (like you got to go back and forth between pages - several times - before you can get anything done). 

Taylor and I have added some really helpful information in the document, and we hope you find it helpful:

image


Sorry for jumping in, but I did not want to miss out on the chance of having the documentation for this element as accurate and clear as possible since I personally know the pain that it is to authenticate anything (programmatically) via Sage One API.

Cheers - Rapha Barbo

**Comment by davewilks** [comment]
@rsbarbo This is great. I'll take a look through and make some updates

File: docs/elements/sageone/authenticate.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+| country | The country code associated with your account at {{page.heading}} |
+| state | A customizable identifier, typically the element key (`{{page.elementKey}}`) . |
+
+{% include note.html content="If the user denies authentication and/or authorization, there will be a query string parameter called <code>error</code> instead of the <code>code</code> parameter. In this case, your application can handle the error gracefully." %}
+
+### Authenticating the Element Instance
+
+Use the `/instances` endpoint to authenticate with {{page.heading}} and create an element instance. If you are configuring events, see the [Events section](events.html).
+
+{% include note.html content="The endpoint returns an Element token upon successful completion. Retain the token for all subsequent requests involving this element instance.  " %}
```
Comment (claytonshaver) [inline docs/elements/sageone/authenticate.md]: We might consider adding "save the instance id" to this as well.

File: docs/elements/sageone/authenticate.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+            }
+    ```
+
+1. Call the following, including the JSON body you constructed in the previous step:
+
+        POST /instances
+
+    {% include note.html content="Make sure that you include the User and Organization keys in the header. See <a href=index.html#authenticating-with-cloud-elements>the Overview</a> for details. " %}
+
+1. Locate the `token` in the response and save it for all future requests using the element instance.
```
Comment (claytonshaver) [inline docs/elements/sageone/authenticate.md]: Same as the note on retaining both the token and the instance id. They need the instance id to delete an instance.

File: docs/elements/sageone/authenticate.md [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+                "key": "{{page.elementKey}}"
+              },
+              "providerData": {
+                "code": "<AUTHORIZATION_GRANT_CODE>"
+              },
+              "configuration": {
+                "oauth.api.key": "<CLIENT_ID>",
+                "oauth.api.secret": "<CLIENT_SECRET>",
+                "signature.secret": "<SIGNING_SECRET>",
+              	"apim.subscription.key": "<SUBSCRIPTION_PRIMARY_KEY_or_APIM_SUBSCRIPTION_KEY>",
```
Comment (claytonshaver) [inline docs/elements/sageone/authenticate.md]: Whats the apim subscription key, is this the grant code from the previous step, or is that different key that we need to pass in?

File: docs/elements/sageone/authenticate.md [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+
+## Authenticate Through API
+
+Authenticating through API is a multi-step process that involves:
+
+* [Getting a redirect URL](#getting-a-redirect-url). This URL sends users to the vendor to log in to their account.
+* [Authenticating users and receiving the authorization grant code](#authenticating-users-and-receiving-the-authorization-grant-code). After the user logs in, the vendor makes a callback to the specified url with an authorization grant code.
+* [Authenticating the element instance](#authenticating-the-element-instance). Using the authorization code from the vendor, authenticate with the vendor to create an element instance at Cloud Elements.
+
+### Getting a Redirect URL
```
Comment (claytonshaver) [inline docs/elements/sageone/authenticate.md]: Just a side note, we should brainstorm how to communicate the authentication flow in a way that shows the series of steps required in order to create that element instance.

File: docs/elements/sageone/authenticate.md (hunk 1 above)
Comment (davewilks) [inline docs/elements/sageone/authenticate.md]: 👍

File: docs/elements/sageone/authenticate.md (hunk 3 above)
Comment (davewilks) [inline docs/elements/sageone/authenticate.md]: It's described in the Parameters table. It's referred to as both APIM subscription key and as Subscription primary key.

File: docs/elements/sageone/authenticate.md (hunk 2 above)
Comment (davewilks) [inline docs/elements/sageone/authenticate.md]: 👍

File: docs/elements/sageone/authenticate.md (hunk 3 above)
Comment (claytonshaver) [inline docs/elements/sageone/authenticate.md]: Is this missing the provider code? Or is that not necessary with this element?

Comment (davewilks) [inline docs/elements/sageone/authenticate.md]: It should be there in line 126

Comment (claytonshaver) [inline docs/elements/sageone/authenticate.md]: yep I missed that, sorry. Looks great!

**Pull request merged**